great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -126,27 +126,32 @@ def _convert_root_connection_detail_fields(cls, values: dict) -> dict:
         for field_name in tuple(values.keys()):
             if field_name in connection_detail_fields:
                 connection_details[field_name] = values.pop(field_name)
+
+        if values.get("connection_string") and connection_details:
+            raise ValueError(
+                "Cannot provide both a connection string or a combination of account, user, and password."
+            )
+
```

Comment:
`test_conflicting_connection_string_and_args_raises_error` was broken. After fixing, found that XOR check wasn't happening, so added this.

@Kilo59 - I assume we still wanted this XOR validation since we still had the above test, and there was a validator called `_check_xor_input_args` (but XOR check wasn't actually happening). LMK if you meant to remove XOR condition.